webui/react/src/pages/TrialDetails/TrialDetailsLogs.tsx

```diff
@@ -179,26 +212,187 @@ const TrialDetailsLogs: React.FC<Props> = ({ experiment, trial }: Props) => {
+    };
+  }, [throttledChangeSearch]);
+
+  const onSearchChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setSearchInput(e.target.value);
+      throttledChangeSearch(e.target.value);
+      if (!e.target.value) {
+        setSearchResults([]);
+        setSelectedLog(undefined);
+      }
+    },
+    [throttledChangeSearch],
+  );
+
+  const formatedSearchResults = useMemo(() => {
+    const key = settings.searchText;
+
+    if (!key) return;
+    const formated: ViewerLog[] = [];
+    searchResults.forEach((l) => {
+      const content = l.log;
+      if (!content) return;
+      if (settings.enableRegex) {
+        try {
+          new RegExp(key);
+        } catch {
+          return;
+        }
+      }
+
+      const logEntry = formatLogEntry(l);
+
+      const i = settings.enableRegex ? content.match(`${key}`)?.index : content.indexOf(key);
+      if (_.isUndefined(i) || i < 0) return;
+      const keyLen = settings.enableRegex ? content.match(`${key}`)?.[0].length || 0 : key.length;
+      const j = i + keyLen;
+      formated.push({
+        ...logEntry,
+        message: `${content.slice(0, i)}<span style="background-color: #E7F7FF">${content.slice(i, j)}</span>${content.slice(j)}`,
+      });
+    });
+    return formated;
+  }, [searchResults, settings.searchText, settings.enableRegex]);
+
+  useEffect(() => {
+    if (settings.searchText) {
+      setSearchResults([]);
+      setSelectedLog(undefined);
+      readStream(
+        handleFetch(
+          {
+            canceler: canceler.current,
+            fetchDirection: FetchDirection.Newer,
+            limit: 0,
+          },
+          FetchType.Initial,
+          settings.searchText,
+          settings.enableRegex,
+        ),
+        (log) => setSearchResults((prev) => [...prev, mapV1LogsResponse(log)]),
+      );
+    }
+  }, [settings.searchText, handleFetch, settings.enableRegex, canceler]);
+
+  const renderSearch = useCallback(() => {
+    const height = container.current?.getBoundingClientRect().height || 0;
+    return (
+      <div className={css.search} style={{ height: `${height - 74}px` }}>
+        <div className={css.header}>
+          <Input
+            allowClear
+            placeholder="Search Logs..."
+            value={searchInput || settings.searchText}
+            onChange={onSearchChange}
+          />
+          <Button onClick={() => setSearchOn(false)}>
+            <Icon decorative name="close" />
+          </Button>
+          {!logViewerOn && (
+            <Button onClick={() => setLogViewerOn(true)}>
+              <Icon decorative name="list" />
+            </Button>
+          )}
+        </div>
+        <Checkbox
+          checked={settings.enableRegex}
+          onChange={(e) => updateSettings({ enableRegex: e.target.checked })}>
+          Regex
+        </Checkbox>
+        <div className={css.logContainer}>
+          {formatedSearchResults && formatedSearchResults.length > 0 ? (
```

Comment:
if we ensure `formatedSearchResults` is always an array, we can get rid of the null check:
```suggestion
          {formatedSearchResults.length > 0 ? (
```